src/UI/PersonNewsFeedBar.jsx

```diff
@@ -39,26 +65,42 @@ const PreferencesForm = () => {
     const sourcesString = activeSources.join(",");
 
     navigate(`/personalized-news/${sourcesString}`);
+    console.log("handle save");
+    toggleModal();
   };
 
+  if (!isOpen) return null;
+
   return (
     <div>
-      <h2>Set Your Preferences</h2>
-      {error && <div style={{ color: "red" }}>{error}</div>}
-      <div>
-        <h3>Sources:</h3>
-        {Object.keys(sources).map((source) => (
-          <label key={source}>
-            <input
-              type="checkbox"
-              checked={sources[source]}
-              onChange={() => handleSourceChange(source)}
-            />
-            {source}
-          </label>
-        ))}
-      </div>
-      <button onClick={handleSave}>Save Preferences</button>
+      <Dialog open={isOpen} PaperProps={{ ref: paperRef }}>
+        <DialogTitle>Set Your Preferences</DialogTitle>
+        {error && (
+          <DialogContentText style={{ color: "red" }}>
+            {error}
+          </DialogContentText>
+        )}
+        <DialogContent>
+          <DialogContentText>Sources:</DialogContentText>
+          <DialogActions>
+            {Object.keys(sources).map((source) => (
+              <label key={source}>
+                <Checkbox
+                  type="checkbox"
+                  checked={sources[source]}
+                  onChange={() => handleSourceChange(source)}
+                />
+                {source}
+              </label>
+            ))}
+          </DialogActions>
+        </DialogContent>
+        <DialogActions>
```

Comment:
Here, I'd suggest adding `Cancel` button, which is always a good user experience, it would be something like this:

```ts
<Button variant="outlined" color="error" onClick={ontoggle}>
  Cancel
</Button>
```